cmd/clusterctl/test/e2e/config_cluster_test.go

```diff
@@ -0,0 +1,130 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	clusterctlclient "sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client"
+	clusterctlrepo "sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client/repository"
+	"sigs.k8s.io/cluster-api/test/framework/management/kind"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("clusterctl config cluster", func() {
+	Describe("basic", func() {
+		var (
+			cfgFile                string // default is $HOME/clusterctl.yaml
+			kindCluster            *kind.Cluster
+			kindClient             client.Client
+			clusterName            string
+			kubernetesVersion      string
+			coreProvider           string
+			coreVersion            string
+			bootstrapProvider      string
+			bootstrapVersion       string
+			infrastructureProvider string
+			infrastructureVersion  string
```

Comment:
We could collapse these into their respective single provider:version variable. 
Eg, `var coreProvider string` could just hold the value `cluster-api:v0.3.0`.